store/migrations/migrate.go

```diff
@@ -0,0 +1,85 @@
+package migrations
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/logger"
+	"github.com/smartcontractkit/chainlink/store/migrations/migration1536521223"
+	"github.com/smartcontractkit/chainlink/store/migrations/migration1536696950"
+	"github.com/smartcontractkit/chainlink/store/orm"
+)
+
+func init() {
+	registerMigration(migration1536521223.Migration{})
```

Comment:
What's your thoughts on doing this at runtime? I was thinking you could read the list of migration directories from disk, get access to the migration in the package and then cast it to the interface through something like:

```golang
// loop through dirs on disk to get path
path := "github.com/smartcontractkit/chainlink/store/migrations/migration1536696950"
pkg, err := importer.Default().Import(path)
registerMigration (migration(pkg.Scope().Lookup("Migration")))
```

Reply:
Reflection techniques are trickier in go than in other languages. An illustration of this is the fact that we don't have any migration directories to iterate over after compilation when we're running from our single go binary.

We tried having the `init` in the individual migration (`migrations/migration123`) self register, but if they're never imported in a top level pkg (`migrations`, or `store`) they are culled out and never get `init`d. The most common pattern we found was the explicit registration in an `init`. Let's chat more offline.

Reply:
Ah, good point about the dirs not being available after compilation. If it's too difficult or requires more research I'm ok with this approach.